packages/node/test/integrations/undici.test.ts

```diff
@@ -0,0 +1,319 @@
+  });
+
+  it('creates a span with internal errors', async () => {
+    const transaction = hub.startTransaction({ name: 'test-transaction' }) as Transaction;
+    hub.getScope().setSpan(transaction);
+
+    try {
+      await fetch('http://a-url-that-no-exists.com');
+    } catch (e) {
+      // ignore
+    }
+
+    expect(transaction.spanRecorder?.spans.length).toBe(2);
+
+    const span = transaction.spanRecorder?.spans[1];
+    expect(span).toEqual(expect.objectContaining({ status: 'internal_error' }));
+  });
+
+  it('does not create a span for sentry requests', async () => {
+    const transaction = hub.startTransaction({ name: 'test-transaction' }) as Transaction;
+    hub.getScope().setSpan(transaction);
+
+    try {
+      await fetch(`${SENTRY_DSN}/sub/route`, {
+        method: 'POST',
+      });
+    } catch (e) {
+      // ignore
+    }
+
+    expect(transaction.spanRecorder?.spans.length).toBe(1);
+  });
+
+  it('does not create a span for sentry requests', async () => {
+    const transaction = hub.startTransaction({ name: 'test-transaction' }) as Transaction;
+    hub.getScope().setSpan(transaction);
+
+    expect(transaction.spanRecorder?.spans.length).toBe(1);
+
+    try {
+      await fetch(`${SENTRY_DSN}/sub/route`, {
+        method: 'POST',
+      });
+    } catch (e) {
+      // ignore
+    }
+
+    expect(transaction.spanRecorder?.spans.length).toBe(1);
+  });
+
+  it('does not create a span if there is no active spans', async () => {
+    try {
+      await fetch(`${SENTRY_DSN}/sub/route`, { method: 'POST' });
+    } catch (e) {
+      // ignore
+    }
+
+    expect(hub.getScope().getSpan()).toBeUndefined();
+  });
+
+  it('does create a span if `shouldCreateSpanForRequest` is defined', async () => {
+    const transaction = hub.startTransaction({ name: 'test-transaction' }) as Transaction;
+    hub.getScope().setSpan(transaction);
+
+    const client = new NodeClient({ ...DEFAULT_OPTIONS, shouldCreateSpanForRequest: url => url.includes('yes') });
+    hub.bindClient(client);
+
+    await fetch('http://localhost:18099/no', { method: 'POST' });
+
+    expect(transaction.spanRecorder?.spans.length).toBe(1);
+
+    await fetch('http://localhost:18099/yes', { method: 'POST' });
+
+    expect(transaction.spanRecorder?.spans.length).toBe(2);
+  });
+
+  it('attaches the sentry trace and baggage headers', async () => {
+    const transaction = hub.startTransaction({ name: 'test-transaction' }) as Transaction;
+    hub.getScope().setSpan(transaction);
+
+    await fetch('http://localhost:18099', { method: 'POST' });
+
+    expect(transaction.spanRecorder?.spans.length).toBe(2);
+    const span = transaction.spanRecorder?.spans[1];
+
+    expect(requestHeaders['sentry-trace']).toEqual(span?.toTraceparent());
+    expect(requestHeaders['baggage']).toEqual(
+      `sentry-environment=production,sentry-transaction=test-transaction,sentry-public_key=0,sentry-trace_id=${transaction.traceId},sentry-sample_rate=1`,
+    );
+  });
```

Comment:
l: Let's add a test to cover the case where `shouldCreateSpanForRequest` returns `false` which should lead to the tracing headers not being attached.